devtools/x-core/src/lib.rs

```diff
@@ -87,8 +87,10 @@ impl XCoreContext {
     }
 
     /// Returns the Git CLI for this workspace.
-    pub fn git_cli(&self) -> &GitCli {
-        &self.git_cli
+    pub fn git_cli(&self) -> Result<&GitCli> {
+        let root = self.project_root;
+        self.git_cli
+            .get_or_try_init(|| -> Result<GitCli> { GitCli::new(root) })
```

Comment:
Was the type annotation required?

Reply:
Compilation Error without it.   Tried without the braces as well.   Honestly can't tell you more than the compiler told me to do it. 🤷

Reply:
```
error: expected type, found `{`
  --> devtools/x-core/src/lib.rs:93:36
   |
93 |             .get_or_try_init(|| -> { GitCli::new(root) } )
   |                                    ^ expected type
   ```

Reply:
oh you can skip both the arrows and the braces.

Reply:
Yeah figured that out a little bit back.   Pr has that fix in it now.